src/resource_server/utils/common.py

```diff
@@ -22,7 +22,7 @@ def paramiko_establish_connection(base_url: str, user: str, host: str, port: int
         ssh
     """
     ssh = paramiko.SSHClient()
-    keys = get_keys(base_url)
+    keys = get_keys(base_url, headers)
```

Comment:
It might be better to factor get_keys out , and pass keys in as parameter instead of headers.
keys is more relevant to this function than headers.